### PR TITLE
Implement adjusting colors for locked points

### PR DESCRIPTION
**Pull request by nishasy**
## Summary:
Now that the interactive graph editor can add locked
points, we also need to make it so that they can update
the color of those points.

In this PR, I added a dropdown that allows for changing
the color. I set the default color to blue. I also
made is so that the actual color value isn't saved
as part of the locked figure styles, just the name.
This then gets converted to the wonder blocks color
later in the mafs graph. I did it this way to avoid
having to convert back and forth between the color name
and value in the color UI.

Issue: https://khanacademy.atlassian.net/browse/LEMS-1751

## Test plan:
`yarn jest packages/perseus-editor/src/widgets/__tests__/interactive-graph-editor.test.tsx`
`yarn jest packages/perseus/src/widgets/__tests__/interactive-graph.test.ts`

Storybook
- Go to http://localhost:6006/?path=/story/perseus-widgets-interactive-graph--segment-with-mafs-and-locked-points
- Confirm that the locked points are blue by default
- Go to http://localhost:6006/?path=/story/perseus-editor-widgets-interactive-graph-editor--with-locked-points
- There should be two locked points: one blue and one gold
- Confirm that using the color dropdown changes the color for the appropriate point
- Add a new point
- Confirm that it's blue by default
- Change its color and confirm all the colors work
- Change the coordinates of the points and make sure that nothing breaks
- Delete points and make sure nothing breaks
- Add points again and make sure nothing breaks 
- Click on the "Color" label and confirm that it opens the color dropdown.
- ^ Repeat for each dropdown and make sure it opens the correct one (IDs are unique and work as expected)

## Demo:

Screenshot 2024-03-14 at 3 19 08 PM

https://github.com/Khan/perseus/assets/13231763/6c8306b1-8364-4096-bfb7-4994db3a33e7

### CODE REVIEW

File: packages/perseus-editor/src/components/locked-point-settings.tsx [hunk 1]
```diff
@@ -18,13 +19,24 @@ import {getValidNumberFromString} from "./util";
...
 
+const possibleColors = [
+    "purple",
+    "blue",
+    "teal",
+    "green",
+    "gold",
+    "red",
+    "pink",
+];
```
Comment (nishasy) [inline packages/perseus-editor/src/components/locked-point-settings.tsx:30]: These are not the exact colors from the [designs](https://www.figma.com/file/JroPokeCLBv2VRIw3GPvY5/LC---Interactive-graph-widget?type=design&node-id=739-6745&mode=design&t=9dixLbZO1PneX7r0-0), but I'm not sure where those are? I did my best to match them with Wonder Blocks colors.

**Review by nedredmond — APPROVED** [review]
I took a few looks over to see if I could find anything objectionable so you wouldn't call me out for green-stamping 😛, but this change legitimately LGTM!

File: packages/perseus-editor/src/components/locked-point-settings.tsx (hunk 1 above)
Comment (jeremywiebe) [reply packages/perseus-editor/src/components/locked-point-settings.tsx:30]: I think there's a similar list defined in the `mathjax-renderer` repo.  You might want to work with Ben and/or Tamara to see if that list should be defined somewhere centrally?

File: packages/perseus/src/widgets/interactive-graphs/graph-locked-layer.tsx [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
+/**
+ * Takes a style object with color names and converts them to actual
+ * Wonder Blocks color values. This way, we can use colors and their
+ * names in the editor UI without having to convert them back and forth.
+ */
+function convertStyle(style?: StyleParams): StyleParams {
+    return {
+        ...style,
+        stroke: color[style?.stroke || "blue"],
+        fill: color[style?.fill || "blue"],
```
Comment (jeremywiebe) [inline packages/perseus/src/widgets/interactive-graphs/graph-locked-layer.tsx:21]: I wonder if we should have a `DefaultColor` variable somewhere?  🤔   (Or `DefaultPointColor` as we'll likely have other locked features that need a default color).

**Pull request merged**